src/cursor/cur_metadata.c

```diff
@@ -93,12 +95,25 @@ __schema_create_collapse(WT_SESSION_IMPL *session, WT_CURSOR_METADATA *mdc, cons
         }
         WT_RET_NOTFOUND_OK(ret);
 
+        if (((ret = __wt_config_getones(session, value, "shared", &cval)) == 0) && cval.val)
+            tiered_shared = true;
+        WT_RET_NOTFOUND_OK(ret);
+
         c = mdc->create_cursor;
         WT_ERR(__wt_scr_alloc(session, 0, &buf));
-        /*
-         * When a table is created without column groups, we create one without a name.
-         */
-        WT_ERR(__wt_buf_fmt(session, buf, "colgroup:%s", key));
+        if (tiered_shared) {
+            active = true;
+tiered_shared:
+            /* When a tiered storage shared table is created, we create two column groups. */
+            if (active)
+                WT_ERR(__wt_buf_fmt(session, buf, "colgroup:%s.%s", key, "active"));
+            else
+                WT_ERR(__wt_buf_fmt(session, buf, "colgroup:%s.%s", key, "shared"));
+        } else {
+            /* When a table is created without column groups, we create one without a name. */
+            WT_ERR(__wt_buf_fmt(session, buf, "colgroup:%s", key));
```

Comment:
Maybe this code could be refactored along with code in `_wt_schema_tiered_shared_colgroup_name()` so there's a single place that knows how to build a column group name.

Reply:
Yes, it is a nice suggestion, done the refactoring.